tests/test-data/test_configs/example_recursor.toml

```diff
@@ -37,4 +37,4 @@ zone_type = "Hint"
 
 ## remember the port, defaults: 53 for Udp & Tcp, 853 for Tls and 443 for Https.
 ##   Tls and/or Https require features dns-over-tls and/or dns-over-https
-stores = { type = "recursor", roots = "default/root.zone", ns_cache_size = 1024, record_cache_size = 1048576 }
+stores = { type = "recursor", roots = "default/root.zone", ns_cache_size = 1024, record_cache_size = 1048576, recursion_limit = 12 }
```

Comment:
My recommendation is to lower the value of `recursion_limit` (for example 2), so it requires less setup in the test.

Reply:
That would simplify the test, but it would also eliminate measurement of one of the test objectives: ensuring that long CNAME chains can be resolved.  I agree the test is quite verbose, but I'd rather work on improving the dns-test API's support for complex tests than eliminate a valid test case to work around the present contour of the API.